### ISSUE TITLE
logical_not of most non-zero value doesn't return 0

### ISSUE THREAD

**Issue by Samuel-DEVULDER** [spotlessmind1975/ugbasic#215]
https://github.com/spotlessmind1975/ugbasic/blob/e11182d3958f245f493ef11692e02816786d6f2d/ugbc/src/hw/6809.c#L2955

I can see that the logical "not" is implemented in term of bit-inversion. This mean (as indicated in the doc) that $00 becomes $FF (and vice versa), but this also mean that non-zero value which are presumed true, will remain true: 
```
IF $AA THEN : PRINT "$AA = TRUE" : ELSE : PRINT "$AA = FALSE" : ENDIF
IF NOT $AA THEN : PRINT "NOT $AA = TRUE" : ELSE : PRINT "NOT $AA = FALSE" : ENDIF
```
result in
```
$AA = TRUE
NOT $AA = TRUE
```
which is very strange when dealing with logical values.

**Comment by spotlessmind1975** [comment]
@Samuel-DEVULDER **excellent observation!** 
In fact I hadn't thought about this side effect.

The reason why I chose this approach depended on the fact that I tried to reduce the true / false logical management into a mathematics operations, with signed number on 2-complement. This made expressions more streamlined, in terms of generated assembly code, as well as the ability to compile things like this:

      POKE 1024, NOT( PEEK( 42 ) OR 3 )
      IF a OR NOT b THEN

with the very same code. 

Probably the solution passes through a differentiation of the `NOT`, if used for logical operations or for bitwise operations. In the logical case it should be rewritten like this (in pseudo assembly code):

        LDA value
        BEQ becomeTrue
        CLR result
        JMP finished
    becomeTrue
        LDA #$FF
        STA result
    finished

Let me know your opinion (or throw me some vegetables if I was really wrong!).

**Comment by Samuel-DEVULDER** [comment]
Personally  on the 6809, I would code the logical "not" like this:
```
    LDA value
    BNE becomeFalse
    LDA #$FF
    FCB $81   ; pseudo SKIP1 opcode
becomeFalse
   CLRA
   STA result
```
It uses a pseudo instruction to jump over a single byte faster than a JMP/BRA (this is actually interpreted as "CMPA #<byte for CLRA>" when the flow comes from "LDA #$FF". I have made a few of these optim in my [6809optim branch](https://github.com/Samuel-DEVULDER/ugbasic/commit/d4b9cf1b53132efda746fa110bde32daf8ea8fd9). 

Other variations are possible:
```
    CLRA
    LDB value   ; LDB is faster that TST. Use LDX for 16 bits.
    BNE becomeFalse
    DECA        ; A=$FF
becomeFalse
    STA result
```
(this one ma be more easier to read than the SKIP1 method.. hmm I might revisit my ASM optims)

Now more generally, when reading the [documentation](https://retroprogramming.iwashere.eu/ugbasic:user:maths) about maths it is said that AND and OR are logical-only operators, so it look natural that NOT be the same (logical negation). But then there is no room for bitwise operations which are fairly frequent and useful. May be you can differentiate bitwise operations by suffixing them with "B" (for Bitwise/Binary/Bytes) --> ANDB, ORB, NOTB, XORB, e.g. `POKE 1024, NOTB( PEEK( 42 ) ORB 3 )`.

**Comment by spotlessmind1975** [comment]
@Samuel-DEVULDER given that I completely agree with the latter optimization, it must be said that I do not want to overturn the syntax of BASIC too much. In the sense that this BASIC (ugBASIC) should be similar to what exists on other systems, where there is no distinction between (i.e.) "logical NOT" and "bitwise NOT", or rather the distinction is made at a contextual level, therefore without the introduction of dedicated keywords. 

![image](https://user-images.githubusercontent.com/11718606/141677353-d79b6a13-935c-43af-a316-f41635cc0e62.png)
![image](https://user-images.githubusercontent.com/11718606/141677527-740a18c1-33e4-43bf-a5a9-7cd367669980.png)

For example, this is the example you kindly proposed to me, rewritten in BASIC V2 (the embedded basic in C=64) and in BASIC 128 (the embedded basic in Olivetti Prodest PC128). As you will notice, it actually gives the same result that you indicated as "anomalous". I don't know how we could achieve this, but I think it would be for the best to maintain a backward compatibility with the original BASIC assumptions.

Let me know, I am open to any further consideration.

**Comment by Samuel-DEVULDER** [comment]
Yes indeed.. lets keep with this behavior consistant with other basics. 

May be the user shall be informed in the doc not use "NOT" in expressions to express "is different from 0" as the NOt only does bitwise complement. He should prefer "<>0" instead.

Side note: looking in the [GFA-basic documentation](http://freddo.chez.com/GfaBasic/GFABASIC.TXT) (a basic that I liked upto a point that I did kind of my own version [for the TO9](https://forum.system-cfg.com/viewtopic.php?p=164716&sid=4cc4b36074003b3cba1558d9fee0a293#p164716)), I found that they use the "infix" notation: x AND y to express the logical "and" of non-zero values, and the prefix notation (eg fonction): AND(x,y) to express the bitwise operation. I find this very simple and natural.
